wallet-new/src/Cardano/Wallet/Kernel/DB/HdWallet/Update.hs

```diff
@@ -32,7 +45,13 @@ updateHdRootName rootId name =
 
 updateHdAccountName :: HdAccountId
                     -> AccountName
-                    -> Update' HdWallets UnknownHdAccount ()
-updateHdAccountName accId name =
-    zoomHdAccountId identity accId $
-      hdAccountName .= name
+                    -> Update' HdWallets UpdateHdAccountError HdAccount
+updateHdAccountName accId name = do
+    -- Check that the parent root & account does exist.
+    zoomHdRootId UpdateHdAccountUnknownRoot (accId ^. hdAccountIdParent) $
+      return ()
```

Comment:
This is not necessary. The zoom functions already take care of this.

Reply:
Ah, you would be surprised but I had a failing test due to this, because in case the account one wanted to delete was not there, the function seemed to silently ignore the fact the wallet was not there in the first place. This was fixed by introducing this check.

Reply:
Take a look at the code of `zoomHdAccountId`.